core/client/fs/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -126,13 +147,30 @@ public int read(byte[] b, int off, int len) throws IOException {
 
     int bytesLeft = len;
     int currentOffset = off;
+    int maxRetry = Configuration.getInt(PropertyKey.USER_BLOCK_WORKER_CLIENT_READ_RETRY);
+    int retryLeft = maxRetry;
+    boolean retryCurrentWorker = true;
+    Set<WorkerNetAddress> lostWorkers = new HashSet<>();
```

Comment:
How about making `lostWorkers` a member variable of this class?
So the lost workers can be seen per fileinstream

Reply:
Would FileInStream stay long enough and the lostWorkers become out of date and prevent client from reading from recovered workers?